content/en/docs/papers/fsep/_index.md

```diff
@@ -0,0 +1,218 @@
+
+The obvious issue is that every site can have its blocklists, which is excellent for individual setups. However, this makes collaboration between these lists challenging, which can hamper coordination around dealing with abusive sites that perpetually engage in harassment and bad faith engagement. 
+
+To prioritize this [collaboration](https://codeberg.org/oliphant/blocklists) and make it easier to defederate with sites with a  history of poor behavior, The Bad Space was created, a searchable database populated with sites that multiple instances have blocked. While still in alpha, this site provides new opportunities to enhance safety and abuse mitigation and improve onboarding and blocklist maintenance by providing an easy way to keep blocklists up to date from a central location through dynamically created export lists and public API that can be used to search the database programmatically.
+
+These features can be integrated into existing platforms to enhance the overall experience on Activity Pub-enabled applications by automatically limiting the opportunity for bad-faith actors to interact.  
+
+
+## Functional Requirements
+
+### Service Feature Integration 
+
+The first step in this process is to make defederated systems aware of the features provided by services such as The Bad Space, specifically its public search API and dynamically created and updated blocklists. 
+
+Federation is the exchange of messages through a common language, in this case, Activity Pub. A note is created using a specific format, and platforms that understand that format can read and respond to the original message. When this connection happens, sites are officially federated with each other.
+
+![federation process](federation-visual.png)
+
+Currently, federation can be prevented by adding a URL to the platform’s blocklist,  which will, in turn, refuse to accept activity from the site requesting communication, disallowing federation to happen.  
+
+
+While serviceable, this process has several challenges, such as: 
+
+- Exposing users to harmful content as means of filtering. 
+- Each problematic site must be manually added 
+- No mechanism for automation, which requires blocklists to be manually  maintained and updated 
+- No tool to validate connection requests at the user level for harmful content
+- Underdeveloped blocklist maintenance options to fine-tune what sites can communicate based on the administrator's discretion.
+
+Integration of The Bad Space’s features into backend functionality will improve  not only the administration experience but the individual users as well by: 
+
+- Allowing blocklists to be automatically imported during onboarding dramatically reduces the opportunity to be exposed to harmful content unnecessarily. 
+- The ability to request an updated blocklist or automate the process to check on its own periodically. 
```

Comment:
Aside: Would be interested in whether these can be achieved using existing APIs or if additional work is needed. 

I'm currently working on an PR to add webhooks to domain blocks in Mastodon, and [have also asked about](https://discord.com/channels/231908446830723072/1061480803344855140/1135549584668635137) (on discord) the ability to learn whether a user account signed in via OAuth 2.0 has moderation abilities on that instance. These may help with your endeavours.

Reply:
Specific implementation is outside the scope of the doc.